File: ui/src/app/shared/jsonrpc/request/getUserInformationRequest.ts
import { JsonrpcRequest } from "../base";

/**
 * <pre>
 * {
 *  "jsonrpc": "2.0",
 *  "id": UUID,
 *  "method": "getUserInformation"
 *  }
 * </pre>
 */
export class GetUserInformationRequest extends JsonrpcRequest {

    private static METHOD: string = "getUserInformation";

    public constructor() {
        super(GET_USER_INFORMATION_REQUEST.METHOD, {});
    }

}

File: io.openems.backend.common/src/io/openems/backend/common/jsonrpc/request/GetUserInformationRequest.java
package io.openems.backend.common.jsonrpc.request;

import com.google.gson.JsonObject;

import io.openems.common.jsonrpc.base.JsonrpcRequest;

/**
 * Gets the User Information.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": UUID,
 *   "method": "getUserInformation",
 *   "params": {}
 * }
 * </pre>
 */
public class GetUserInformationRequest extends JsonrpcRequest {

	public static final String METHOD = "getUserInformation";

	/**
	 * Create {@link GetUserInformationRequest} from a template
	 * {@link JsonrpcRequest}.
	 *
	 * @param request the template {@link JsonrpcRequest}
	 * @return Created {@link GetUserInformationRequest}
	 */
	public static GetUserInformationRequest from(JsonrpcRequest request) {
		return new GetUserInformationRequest(request);
	}

	private GetUserInformationRequest(JsonrpcRequest request) {
		super(request, GetUserInformationRequest.METHOD);
	}

	@Override
	public JsonObject getParams() {
		return new JsonObject();
	}

}
